tests/e2e/component.go

```diff
@@ -470,4 +472,228 @@ func componentTests(componentCmdPrefix string) {
 			waitForDeleteCmd("odo project list", projName)
 		})
 	})
+
+	var _ = Context("when component is in the current directory", func() {
+
+		//new clean project and context for each test
+		var project string
+		var context string
+
+		//  current directory and project (before eny test is run) so it can restored  after all testing is done
+		var originalDir string
+		var originalProject string
+
+		// Setup up state for each test spec
+		// create new project (not set as active) and new context directory for each test spec
+		// This is before every spec (It)
+		var _ = BeforeEach(func() {
+			project = helper.OcCreateRandProject()
+			context = helper.CreateNewContext()
+		})
+
+		// Clean up after the test
+		// This is run after every Spec (It)
+		var _ = AfterEach(func() {
+			helper.OcDeleteProject(project)
+			helper.DeleteDir(context)
+		})
+
+		// we will be testing components that are created from the current directory
+		// switch to the clean context dir before each test
+		var _ = JustBeforeEach(func() {
+			originalDir = helper.Getwd()
+			helper.Chdir(context)
+		})
+
+		// go back to original directory after each test
+		var _ = JustAfterEach(func() {
+			helper.Chdir(originalDir)
+		})
+
+		var _ = Context("when --context is not used", func() {
+			var _ = Context("when project from KUBECONFIG is used", func() {
+				// Set active project for each test spec
+				var _ = JustBeforeEach(func() {
+					helper.OcSwitchProject(project)
+				})
+				// go back to original project after each test
+				var _ = JustAfterEach(func() {
+					helper.OcSwitchProject(originalProject)
+				})
+
+				It("create local nodejs component and push source and code separately", func() {
+					appName := "nodejs-push-test"
+					cmpName := "nodejs"
+					helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+					helper.CmdShouldPass("odo component create nodejs " + cmpName + " --app " + appName)
```

Comment:
you have this inside `Context("when --context is not used",` but you are not using context!